.dockerignore

```diff
@@ -0,0 +1,47 @@
+# python envs
+venv
+.env
+
+# do not leak credentials
+.docker
+*.sqlite3
+*.db
+*.log
+
+# Byte-compiled / optimized / DLL files
+**/__pycache__
+**/*.pyc
+**/*.pyo
+.eggs
+*.egg-info
+build
+
+# Unit test / coverage reports
+htmlcov
+.tox
+.nox
+.coverage
+.coverage.*
+.cache
+nosetests.xml
+coverage.xml
+*.cover
+*.py,cover
+.hypothesis
+.pytest_cache
+cover
+
+# editor files
+*~
+.*.swp
```

Comment:
note `.dockeringore` syntax is different with `.gitignore`, this means only ignore root level files, can use [gitignore-to-dockerignore](https://www.npmjs.com/package/gitignore-to-dockerignore) to convert all items from `.gitignore` to `.dockerignore`, or just cherry pick ones we need as we don't have `COPY . .`, many root level items can be skipped

Reply:
I've updated the ignore files slightly. Let's keep all the root items just as a fail-safe. it doesn't hurt :)

Reply:
for example `*.pyc` in `.dockerignore` will just ignore `.pyc` files in root, `/python_project_template/core.pyc` will be added if it exists on local

Reply:
Another try in the last commit